mouse.go

```diff
@@ -28,38 +46,216 @@ func (m MouseEvent) String() (s string) {
 	if m.Alt {
 		s += "alt+"
 	}
-	s += mouseEventTypes[m.Type]
+	if m.Shift {
+		s += "shift+"
+	}
+
+	if m.isSGR {
+		if m.Button == MouseButtonNone && m.Action == MouseActionMotion {
+			s += mouseActions[m.Action]
+		} else if m.IsWheel() {
+			s += mouseButtons[m.Button]
+		} else {
+			s += mouseButtons[m.Button]
+			s += " "
+			s += mouseActions[m.Action]
+		}
+	} else {
+		s += mouseEventTypes[m.Type]
+	}
+
 	return s
 }
 
+// MouseAction represents the action that occurred during a mouse event.
+type MouseAction int
+
+// Mouse event actions
+const (
+	MouseActionPress MouseAction = iota
+	MouseActionRelease
+	MouseActionMotion
+)
+
+var mouseActions = map[MouseAction]string{
+	MouseActionPress:   "press",
+	MouseActionRelease: "release",
+	MouseActionMotion:  "motion",
+}
+
+// MouseButton represents the button that was pressed during a mouse event.
+type MouseButton int
+
+// Mouse event buttons
+//
+// This is based on X11 mouse button codes.
+//
+//	1 = left button
+//	2 = middle button (pressing the scroll wheel)
+//	3 = right button
+//	4 = turn scroll wheel up
+//	5 = turn scroll wheel down
+//	6 = push scroll wheel left
+//	7 = push scroll wheel right
+//	8 = 4th button (aka browser backward button)
+//	9 = 5th button (aka browser forward button)
+//
+// Other buttons are not supported.
+const (
+	MouseButtonNone MouseButton = iota
+	MouseButtonLeft
+	MouseButtonMiddle
+	MouseButtonRight
+	MouseButtonWheelUp
+	MouseButtonWheelDown
+	MouseButtonWheelLeft
+	MouseButtonWheelRight
+	MouseButtonBackward
+	MouseButtonForward
+	MouseButton10
+	MouseButton11
+
+	MouseButtonUnknown
+)
+
+var mouseButtons = map[MouseButton]string{
+	MouseButtonNone:       "none",
+	MouseButtonLeft:       "left",
+	MouseButtonMiddle:     "middle",
+	MouseButtonRight:      "right",
+	MouseButtonWheelUp:    "wheel up",
+	MouseButtonWheelDown:  "wheel down",
+	MouseButtonWheelLeft:  "wheel left",
+	MouseButtonWheelRight: "wheel right",
+	MouseButtonBackward:   "backward",
```

Comment:
I'd probably rename this (and the deprecated version) to `MouseButtonBack` so it's super clear that it's the back button and not a backwards motion.